tests/TestCase/Datasource/ConnectionManagerTest.php

```diff
@@ -306,6 +306,31 @@ public function dsnProvider()
                     'log' => '1'
                 ]
             ],
+            'no password' => [
+                'mysql://user@localhost:3306/database',
```

Comment:
Is the password supposed to be '' here or `null`? Don't both turn into `''` when sent to the driver?